File: src/main/java/TimeRangeToDayMap.java
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Objects;
import java.util.Set;

public class TimeRangeToDayMap {

  private final TimeRange timeRange;
  private final Set<String> days;

  public TimeRangeToDayMap(final TimeRange timeRange) {
    this.timeRange = timeRange;
    days = new HashSet<>();
  }

  public void addDay(final String day) {
    days.add(day);
  }

  List<String> getDays() {
    List<String> dayList = new ArrayList<>();
    if (days.contains("Mon")) {
      dayList.add("Mon");
    }
    if (days.contains("Tue")) {
      dayList.add("Tue");
    }
    if (days.contains("Wed")) {
      dayList.add("Wed");
    }
    if (days.contains("Thu")) {
      dayList.add("Thu");
    }
    if (days.contains("Fri")) {
      dayList.add("Fri");
    }

    return dayList;
  }

  public TimeRange getTimeRange() {
    return timeRange;
  }

  public String getDaysString() {
    return String.join(",", getDays());
  }

  @Override
  public boolean equals(final Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    final TimeRangeToDayMap that = (TimeRangeToDayMap) o;
    return timeRange.equals(that.timeRange) &&
        days.equals(that.days);
  }

  @Override
  public int hashCode() {
    return Objects.hash(timeRange, days);
  }

  @Override
  public String toString() {
    return "TimeRangeMap{" +
        "timeRange=" + timeRange +
        ", days=" + getDaysString() +
        '}';
  }
}

File: src/main/java/com/adamkoch/iptables/TimeRangeToDayMap.kt
package com.adamkoch.iptables

import com.adamkoch.annotations.Unstable
import java.util.*

/**
 *
 * @since 0.1.0
 * @author aakoch
 */
@Unstable
class TimeRangeToDayMap(val timeRange: TimeRange) {
    private val days: MutableSet<String>
    fun addDay(day: String) {
        days.add(day)
    }

    fun getDays(): List<String> {
        val dayList: MutableList<String> = ArrayList()
        if (days.contains("Mon")) {
            dayList.add("Mon")
        }
        if (days.contains("Tue")) {
            dayList.add("Tue")
        }
        if (days.contains("Wed")) {
            dayList.add("Wed")
        }
        if (days.contains("Thu")) {
            dayList.add("Thu")
        }
        if (days.contains("Fri")) {
            dayList.add("Fri")
        }
        return dayList
    }

    val daysString: String
        get() = java.lang.String.join(",", getDays())

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as TimeRangeToDayMap
        return timeRange == that.timeRange && days == that.days
    }

    override fun hashCode(): Int {
        return Objects.hash(timeRange, days)
    }

    override fun toString(): String {
        return "TimeRangeMap{" +
                "timeRange=" + timeRange +
                ", days=" + daysString +
                '}'
    }

    init {
        days = HashSet()
    }
}